File: nutrinova-ui/src/app/(authorized)/meals/record/_models/recordMealRequest.ts
// TypeScript type for the MealType enum
export enum MealItemType {
  CustomFood = "CustomFood",
  Recipe = "Recipe",
  USDAFood = "USDAFood",
}

// TypeScript interface for the RecordMealRequest
export interface RecordMealRequest {
  patientId: string;
  amount: number;
  recordedAt: Date; // DateOnly in C# translates to Date in TypeScript
  unitId: number;
  selectedMealItemId: string;
  mealSelectionType: string;
}

File: nutrinova-api/NutrinovaData/Features/Meals/RecordMealRequest.cs
namespace NutrinovaData.Features.Meals;

public class RecordMealRequest
{
  public Guid PatientId { get; set; }

  public decimal Amount { get; set; }

  public DateTime RecordedAt { get; set; }

  public int UnitId { get; set; }

  public Guid SelectedMealItemId { get; set; }

  public string MealSelectionType { get; set; } = null!;
}

public enum MealSelectionItemType
{
  CustomFood,
  Recipe,
  USDAFood,
  ClonedMeal,
}
